org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -505,6 +505,20 @@ private Optional<Either<Command, CodeAction>> getGenerateConstructorsAction(Code
 			if (type == null || type.isAnnotation() || type.isInterface() || type.isAnonymous() || type.getCompilationUnit() == null) {
 				return Optional.empty();
 			}
+
+			boolean allStaticField = true;

```

Comment:
See https://github.com/eclipse/eclipse.jdt.ls/pull/2534#pullrequestreview-1340164222 . There should be a private helper method in this class that does this logic already, and it should simplify this.